owner/manager.go

```diff
@@ -277,11 +288,20 @@ func init() {
 	}
 }
 
-func isContextDone(ctx goctx.Context) bool {
+func contextDone(ctx goctx.Context, err error) error {
 	select {
 	case <-ctx.Done():
-		return true
+		return errors.Trace(ctx.Err())
 	default:
 	}
-	return false
+	// Sometime the ctx isn't closed, but the etcd client is closed,
+	// we need to treat it as if context is done.
+	// TODO: Make sure ctx is closed with etcd client.
```

Comment:
We don't have the cancel function, how can we make sure `ctx` is closed?

Reply:
So I will do it in the future.